### PR TITLE
feat(stripe): handle `customer.subscription.created` webhook event

### PR DESCRIPTION
**Pull request by bytaesu**
This PR changes:

- Handles `customer.subscription.created` webhook to sync subscriptions created outside the checkout flow (e.g. Stripe Dashboard or API)
- Updated Stripe plugin to use `logger` from context instead of global

### To-do
- [x] Verify behavior in the real app setup

---
## Summary by cubic
Add support for the Stripe customer.subscription.created webhook to sync subscriptions created outside checkout with our database and prevent broken subscription states.

- **New Features**
  - Handle customer.subscription.created in the webhook route.
  - Create subscription records for users matched by stripeCustomerId; map plan via priceId/lookup_key; set seats, period, and trial dates.
  - Add subscription.onSubscriptionCreated callback and document it; docs updated to include the event and getCheckoutSessionParams.

- **Bug Fixes**
  - Skip creation if the subscription already exists; guard when user or plan is missing; use request-scoped logging.
  - Add tests for success, duplicate, missing user, and missing plan scenarios.

<sup>Written for commit cf7243d24925a373c90d6aeab6227a5a269d1d37. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/stripe/src/types.ts [hunk 1]
```diff
@@ -257,6 +257,18 @@ export type SubscriptionOptions = {
...
+	/**
+	 * A callback to run when a subscription is created
+	 * @returns
+	 */
+	onSubscriptionCreated?:
+		| ((data: {
+				event: Stripe.Event;
+				stripeSubscription: Stripe.Subscription;
+				subscription: Subscription;
+				plan?: StripePlan;
```
[the remark this answers is not in the document]
Comment (bytaesu) [reply packages/stripe/src/types.ts]: good

**Comment by bytaesu** [comment]
=> I'm improving docs

File: packages/stripe/src/hooks.ts [hunk 2]
```diff
@@ -93,7 +92,124 @@ export async function onCheckoutSessionCompleted(
...
+			});
+		if (existingSubscription) {
+			ctx.context.logger.info(
+				`Stripe webhook: Subscription ${subscriptionCreated.id} already exists in database, skipping creation`,
+			);
+			return;
+		}
+
+		// Find user by stripeCustomerId
+		const user = await ctx.context.adapter.findOne<{ id: string }>({
```
Comment (himself65) [inline packages/stripe/src/hooks.ts]: Use `User` type here

File: packages/stripe/src/hooks.ts [hunk 3]
```diff
@@ -93,7 +92,124 @@ export async function onCheckoutSessionCompleted(
...
+		const priceLookupKey = subscriptionItem.price.lookup_key || null;
+		const plan = await getPlanByPriceInfo(options, priceId, priceLookupKey);
+		if (!plan) {
+			ctx.context.logger.warn(
+				`Stripe webhook warning: No matching plan found for priceId: ${priceId}`,
+			);
+			return;
+		}
+
+		const seats = subscriptionItem.quantity || 1;
```
Comment (himself65) [inline packages/stripe/src/hooks.ts]: allow it to be `undefined`

File: packages/stripe/src/hooks.ts (hunk 2 above)
Comment (bytaesu) [reply packages/stripe/src/hooks.ts]: Thanks

File: packages/stripe/src/hooks.ts (hunk 3 above)
Comment (bytaesu) [reply packages/stripe/src/hooks.ts]: I updated it so the hook accepts `undefined`.

In the routes it's being converted to 1 using `|| 1`. What's the difference between these?

File: packages/stripe/src/hooks.ts [hunk 4]
```diff
@@ -1,5 +1,4 @@
-import type { GenericEndpointContext } from "better-auth";
-import { logger } from "better-auth";
+import type { GenericEndpointContext, User } from "better-auth";
```
Comment (himself65) [inline packages/stripe/src/hooks.ts]: can we import type from `@better-auth/core` ?

Comment (bytaesu) [reply packages/stripe/src/hooks.ts]: https://github.com/bytaesu/better-auth/blob/561f88d88b8d0c9c034df8b8dc18211b74ddc8a8/packages/stripe/src/types.ts#L6

The User type exists in `better-auth`, and other files already import it from there, so it seems cleaner to keep this consistent 🤔

screen

**Comment by bytaesu** [comment]
This PR needs overall updates to align with recent merged changes. I'll prioritize https://github.com/better-auth/better-auth/pull/6961 first.
